### PR TITLE
Fixing issue #21 (GetNextOccurrence throws if next occurrence produces invalid time)

### PR DESCRIPTION
**Pull request by Styxxy**

### CODE REVIEW

File: NCrontab.Tests/CrontabScheduleTests.cs [hunk 1]
```diff
@@ -377,6 +378,42 @@ static void BadField(string expression, bool includingSeconds)
...
+        /// Test case for issue #21
+        /// (GetNextOccurrence throws if next occurrence produces invalid time)
+        /// </summary>
+        [Test]
+        public void GetNextOccurrences_NextOccurrenceInvalidTime_ShouldStopAtLastValidTime()
+        {
+            var schedule = CrontabSchedule.Parse("0 0 29 Feb Mon");
+            var occurrences = schedule.GetNextOccurrences(new DateTime(9988, 1, 1), DateTime.MaxValue);
+
+            var l = occurrences.ToList();
```
Comment (atifaziz) [inline NCrontab.Tests/CrontabScheduleTests.cs]: This doesn't seem to be doing anything as `l` isn't used and we shouldn't be expecting any side-effects. What did you have in mind?

File: NCrontab/CrontabSchedule.cs [hunk 2]
```diff
@@ -337,6 +339,9 @@ public DateTime GetNextOccurrence(DateTime baseTime, DateTime endTime)
             //  Jan 15, Jan 31, Feb 15, Mar 15, Apr 15, Apr 31, ...
             //
 
+            if (year > Calendar.MaxSupportedDateTime.Year)
```
Comment (atifaziz) [inline NCrontab/CrontabSchedule.cs]: Why not move this earlier to [right after where year in incremented (`year++`)](https://github.com/atifaziz/NCrontab/blob/94a93d677541e8f1a3b3af232ae9d32234a43cc6/NCrontab/CrontabSchedule.cs#L313)? It seems a bit disconnected here and unrelated to [the preceding comment](https://github.com/atifaziz/NCrontab/pull/33/commits/a7211277e05a142a08aaecf08b0b5f2776802062#diff-4ed075ffe845773e5039992a428a7b90R325).

File: NCrontab.Tests/CrontabScheduleTests.cs (hunk 1 above)
Comment (Styxxy) [reply NCrontab.Tests/CrontabScheduleTests.cs]: Woops, it was for debugging ;). I'll remove it.

File: NCrontab/CrontabSchedule.cs (hunk 2 above)
Comment (Styxxy) [reply NCrontab/CrontabSchedule.cs]: Yes, I agree with that; I'll move the code.

**Comment by Styxxy** [comment]
@atifaziz Fixed the code (review comments).

**Comment by Styxxy** [comment]
Thanks for updates. Here and there you use different style than I am used to (so I have to get used to it a little bit).

**Comment by atifaziz** [comment]
Squash-merged in f0c43e00e65aa025506229891718460e1ad7916b.

Thanks! :+1:

**Comment by atifaziz** [comment]
@Styxxy Sometimes it's a little easier/quicker to go ahead & make those small changes with direct commits to the PR than a back-and-forth of review comments around styling (which is more important if you are a more regular contributor).

**Comment by Styxxy** [comment]
Ok, I will try to keep that in mind. I'm fairly new to (really) using GitHub and contributing :wink: .

**Comment by Styxxy** [comment]
Ok, I will try to keep that in mind. I'm fairly new to (really) using GitHub and contributing :wink: .

**Comment by atifaziz** [comment]
> I'm fairly new to (really) using GitHub and contributing 

@Styxxy Good luck. And hey, you've made a [contribution to this project](https://github.com/atifaziz/NCrontab/commits?author=Styxxy) in October so 3 more to go and hope you get your [Hacktoberfest](https://hacktoberfest.digitalocean.com/) T-Shirt. 👕

**Pull request closed without merging**